src/shared/sol-util.c

```diff
@@ -312,14 +315,37 @@ sol_util_str_split(const struct sol_str_slice slice, const char *delim, size_t m
 static int
 get_progname(char *out, size_t size)
 {
-    char cwd[PATH_MAX] = { NULL }, *execfn;
+    char cwd[PATH_MAX] = { NULL }, *execfn = NULL;
 
     if (getcwd(cwd, sizeof(cwd)) == NULL)
-        return -1;
+        return -errno;
 
+#ifdef HAVE_SYS_AUXV_H
     execfn = (char *)getauxval(AT_EXECFN);
```

Comment:
If the header does not exist, then `getauxval()` won't exist, and this value won't be needed.